core/domain/user_validators.py

```diff
@@ -95,9 +95,32 @@ def _validate_time_fields_of_user_actions(cls, item):
                 'greater than the time when job was run' % (
                     item.id, item.first_contribution_msec))
 
+    @classmethod
+    def _validate_profile_user_role_is_learner(cls, item):
```

Comment:
I found that we don't have many checks of prod validation for UserSettingModel, I've added one as I wanted to ensure that it's true in the datastore.

/cc @vojtechjelinek

Reply:
Yep, rn we have been focusing more on rewriting the checks to Beam jobs, feel free to add requirements for other checks that you find useful.

Reply:
I'll defer it to a separate PR as I want this PR to be more specific toward the one-off job!